docs/modules/ROOT/pages/how-tos/cloudscale/provider-adopt-worker-nodes.adoc

```diff
@@ -0,0 +1,220 @@
+= Adopt worker nodes with the cloudscale Machine API Provider
+
+[abstract]
+--
+Steps to adopt worker nodes on https://cloudscale.ch[cloudscale] with the https://github.com/appuio/machine-api-provider-cloudscale[cloudscale Machine API Provider].
+--
+
+== Starting situation
+
+* You already have an OpenShift 4 cluster on cloudscale
+* You have admin-level access to the cluster
+* You want the nodes adopted by the https://github.com/appuio/machine-api-provider-cloudscale[cloudscale Machine API Provider]
+
+== Prerequisites
+
+The following CLI utilities need to be available locally:
+
+* `commodore`, see https://syn.tools/commodore/running-commodore.html[Running Commodore]
+* `docker`
+* `kubectl`
+* `vault`
+* `yq`
+
+== Prepare local environment
+
+include::partial$cloudscale/setup-local-env.adoc[]
+
+== Update Cluster Config
+
+. Update cluster config
++
+[source,bash]
+----
+pushd inventory/classes/"${TENANT_ID}"
+
+yq -i '.applications += "machine-api-provider-cloudscale"' \
+  ${CLUSTER_ID}.yml
+
+yq eval -i ".parameters.openshift4_terraform.terraform_variables.make_worker_adoptable_by_provider = true" \
+  ${CLUSTER_ID}.yml
+yq eval -i '.parameters.machine_api_provider_cloudscale.secrets["cloudscale-user-data"].stringData.ignitionCA = "${openshift4_terraform:terraform_variables:ignition_ca}"' \
+  ${CLUSTER_ID}.yml
+
+git commit -m "Allow adoption of worker nodes" "${CLUSTER_ID}.yml"
+popd
+----
+
+. Compile and push the cluster catalog.
++
+[source,bash]
+----
+commodore catalog compile "${CLUSTER_ID}" --push
+----
+
+== Prepare Terraform environment
+
+include::partial$cloudscale/configure-terraform-secrets.adoc[]
+
+include::partial$setup_terraform.adoc[]
+
+== Run terraform
+
+. Verify terraform output and apply the changes if everything looks good.
++
+Terraform will tag the nodes as preparation for the adoption by the cloudscale Machine API Provider.
++
+[source,bash]
+----
+terraform apply
+----
+
+== Apply Machine and MachineSet manifests
+
+[IMPORTANT]
+====
+Please ensure the terraform apply has completed successfully before proceeding with this step.
+Without the tags applied by Terraform, nodes will be duplicated with the same name and weird stuff might happen.
+
+Please be careful to not apply the `MachineSet` before the `Machine` manifests.
+====
+
+. Copy `worker-machines_yml` from the Terraform output and apply it to the cluster.
++
+[source,bash]
+----
+terraform output -raw worker-machines_yml | yq -P > worker-machines.yml
```

Comment:
on my machine Terraform prints a warning which then gets included in the yaml (because they have never heard of stderr):
```
└─» terraform output -raw worker-machines_yml > foo.yml


└─» cat foo.yml 
There are some problems with the CLI configuration:
╷
│ Warning: Unable to open CLI configuration file
│
│ The CLI configuration file at "/root/.terraformrc" does not exist.
╵

"apiVersion": "v1"
"items": []
"kind": "List"

```

Reply:
I ended up using this command:
```
terraform output -raw worker-machineset_yml | grep -vP '^(│|╵|╷|There are some problems with the CLI configuration)' | yq -P > worker-machineset.yml
head worker-machineset.yml
kubectl apply -f worker-machineset.yml
```

Reply:
Added https://github.com/appuio/openshift4-docs/pull/362/files#diff-3e6bd975dce3122e00a6e4e141865df2931c271706cbd8df6f89d60276b7a0d0R16